src/commands/RefreshIfNeeded.ts

```diff
@@ -33,21 +33,28 @@ const refreshAccessToken = async (credentials: Credentials, config: ApiConfigura
         }));
 }
 
-const createRequest = async (request: Request, config: ApiConfiguration, credentialStorage: CredentialStorage): Promise<Request> => {
+const createRefreshRequest = async (request: Request, config: ApiConfiguration, credentialStorage: CredentialStorage): Promise<Request> => {
     let credentials: Credentials|undefined;
     credentials = credentialStorage.getCredentials()
     if (!credentials) {
         throw Error("Misconfiguration: no credentials provided")
     }
-    credentials = await refreshAccessToken(credentials, config);
+
+    if (credentials.refreshToken) {
```

Comment:
We'd need to have a test for that to make sure that this problem won't happen in the future again.